tests/integration/test_integration.py

```diff
@@ -22,45 +22,45 @@
 
 
 class TestVaultK8s:
-    @pytest.fixture(scope="module")
-    @pytest.mark.abort_on_fail
-    async def charm(self, ops_test):
-        ops_test.destructive_mode = False
-        charm = await ops_test.build_charm(".")
-        return charm
+    @staticmethod
+    async def wait_for_load_balancer_address(
+        kubernetes: Kubernetes, timeout: int = 60
+    ) -> Optional[str]:
+        """Waits for LoadBalancer address to be available and returns it.
 
-    @pytest.fixture()
-    async def cleanup(self, ops_test):
-        try:
-            await ops_test.model.remove_application(
-                app_name=APPLICATION_NAME, block_until_done=True
-            )
-        except JujuError:
-            pass
+        Args:
+            kubernetes: Kubernetes object.
+            timeout: Timeout (seconds).
 
-    @staticmethod
-    async def wait_for_load_balancer_address(kubernetes: Kubernetes, timeout: float = 60):
+        Retunrs:
+            str: LoadBalancer address.
+
+        Raises:
+            TimeoutError: Whether LoadBalancer address is not available after timeout.
+        """
         initial_time = time.time()
         while time.time() - initial_time < timeout:
-            load_balancer_address = kubernetes.get_load_balancer_address(
+            if load_balancer_address := kubernetes.get_load_balancer_address(
                 service_name=APPLICATION_NAME
-            )
-            if load_balancer_address:
+            ):
                 return load_balancer_address
             time.sleep(5)
         raise TimeoutError("Timed out waiting for Loadbalancer address to be available.")
 
     @staticmethod
-    async def initialize_vault(vault: Vault, timeout: int = 60) -> Tuple[str, str]:
+    async def initialize_vault(vault: Vault, timeout: int = 60) -> Optional[Tuple[str, str]]:
         """Initializes Vault.
 
         Args:
             vault: Vault object.
-            timeout: Timeout (seconds)
+            timeout: Timeout (seconds).
 
         Returns:
-            str: Vault's Unseal key
-            str: Vault's Root token
+            str: Vault's Unseal key.
+            str: Vault's Root token,
+
+        Raises:
+            TimeoutError: Whether Vault is not ready after timeout.
```

Comment:
I might be wrong again, but I feel like `Whether` is good when you describe a boolean value. Here I'd rather use `When`. What do you think?

Reply:
You're right, changed.